Installation of labextension fails on Windows 10

I tried to install by following the directions: 

    $ conda install ipympl -c conda-forge
    $ jupyter labextension install jupyter-matplotlib

This is what I got:

```
λ conda install ipympl -c conda-forge
Fetching package metadata .................
Solving package specifications: .

# All requested packages already installed.
# packages in environment at C:\Users\Jeremy\Anaconda3:
#
ipympl
```
and then this
```
λ jupyter labextension install jupyter-matplotlib
> npm.cmd pack jupyter-matplotlib
jupyter-matplotlib-0.1.0.tgz
> node node-version-check.js
> npm.cmd install
-
> jupyter-matplotlib@0.1.0 postinstall C:\Users\Jeremy\Anaconda3\share\jupyter\lab\staging\node_modules\jupyter-matplotlib
> npm run clean && npm run build


> jupyter-matplotlib@0.1.0 clean C:\Users\Jeremy\Anaconda3\share\jupyter\lab\staging\node_modules\jupyter-matplotlib > rimraf dist/

'rimraf' is not recognized as an internal or external command,
operable program or batch file.

npm ERR! Windows_NT 10.0.15063
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "clean"
npm ERR! node v4.4.7
npm ERR! npm  v2.15.8
npm ERR! code ELIFECYCLE
npm ERR! jupyter-matplotlib@0.1.0 clean: `rimraf dist/`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the jupyter-matplotlib@0.1.0 clean script 'rimraf dist/'.
npm ERR! This is most likely a problem with the jupyter-matplotlib package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     rimraf dist/
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs jupyter-matplotlib
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!
npm ERR!     npm owner ls jupyter-matplotlib
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\Jeremy\Anaconda3\share\jupyter\lab\staging\node_modules\jupyter-matplotlib\npm-debug.log
npm ERR! Windows_NT 10.0.15063
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v4.4.7
npm ERR! npm  v2.15.8
npm ERR! code ELIFECYCLE

npm ERR! jupyter-matplotlib@0.1.0 postinstall: `npm run clean && npm run build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the jupyter-matplotlib@0.1.0 postinstall script 'npm run clean && npm run build'.
npm ERR! This is most likely a problem with the jupyter-matplotlib package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run clean && npm run build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs jupyter-matplotlib
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!
npm ERR!     npm owner ls jupyter-matplotlib
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\Jeremy\Anaconda3\share\jupyter\lab\staging\npm-debug.log
> npm.cmd run build

> @jupyterlab/application-top@0.10.2 build C:\Users\Jeremy\Anaconda3\share\jupyter\lab\staging
> webpack

Hash: 998f9dcdfce9168b95b4
Version: webpack 2.7.0
Time: 18682ms
                                 Asset     Size  Chunks                    Chunk Names
  674f50d287a8c48dc19ba404d20fe713.eot   166 kB          [emitted]
af7ae505a9eed503f8b8e6982036873e.woff2  77.2 kB          [emitted]
 fee66e712a8a08eef5805a46892932ad.woff    98 kB          [emitted]
  b06871f281fee6b241d60582ae9369b9.ttf   166 kB          [emitted]
  912ec66d7572ff821749319396470bde.svg   444 kB          [emitted]  [big]
                           0.bundle.js   894 kB       0  [emitted]  [big]
                        main.bundle.js  7.78 MB       1  [emitted]  [big]  main
                       0.bundle.js.map  1.09 MB       0  [emitted]
                    main.bundle.js.map  9.42 MB       1  [emitted]         main
  [24] ./~/@jupyterlab/application/lib/index.js 5.48 kB {1} [built]
 [488] ./~/@jupyterlab/application-extension/lib/index.js 6.12 kB {1} [optional] [built]
 [489] ./~/@jupyterlab/apputils-extension/lib/index.js 7.91 kB {1} [optional] [built]
 [490] ./~/@jupyterlab/codemirror-extension/lib/index.js 10.5 kB {1} [optional] [built]
 [491] ./~/@jupyterlab/completer-extension/lib/index.js 7.09 kB {1} [optional] [built]
 [492] ./~/@jupyterlab/console-extension/lib/index.js 12.5 kB {1} [optional] [built]
 [493] ./~/@jupyterlab/csvviewer-extension/lib/index.js 1.79 kB {1} [optional] [built]
 [494] ./~/@jupyterlab/docmanager-extension/lib/index.js 10.1 kB {1} [optional] [built]
 [495] ./~/@jupyterlab/faq-extension/lib/index.js 3.78 kB {1} [optional] [built]
 [496] ./~/@jupyterlab/filebrowser-extension/lib/index.js 11.9 kB {1} [optional] [built]
 [497] ./~/@jupyterlab/fileeditor-extension/lib/index.js 11.8 kB {1} [optional] [built]
 [498] ./~/@jupyterlab/help-extension/lib/index.js 8.84 kB {1} [optional] [built]
 [499] ./~/@jupyterlab/imageviewer-extension/lib/index.js 4.25 kB {1} [optional] [built]
 [500] ./~/@jupyterlab/inspector-extension/lib/index.js 7.09 kB {1} [optional] [built]
 [529] ./build/index.out.js 8.36 kB {1} [built]
    + 1309 hidden modules

WARNING in ./build/index.out.js
Module not found: Error: Can't resolve 'jupyter-matplotlib/src/labplugin' in 'C:\Users\Jeremy\Anaconda3\share\jupyter\lab\staging\build'
 @ ./build/index.out.js 230:37-80
```


I was going to attach the `npm-debug.log`  file that was mentioned in the error message. However, I couldn't find it at the path that was mentioned. Any ideas on how I fix this? I'm on Windows 10. I'm using Anaconda with Python 3.6.


EDIT: 

It turns out that it seemed to work after I installed npm rimraf.

```
λ npm install rimraf -g
C:\Users\Jeremy\AppData\Roaming\npm\rimraf -> C:\Users\Jeremy\AppData\Roaming\npm\node_modules\rimraf\bin.js
rimraf@2.6.2 C:\Users\Jeremy\AppData\Roaming\npm\node_modules\rimraf
└── glob@7.1.2 (path-is-absolute@1.0.1, inherits@2.0.3, fs.realpath@1.0.0, once@1.4.0, inflight@1.0.6, minimatch@3.0.4)
```

Is this supposed to be a normal part of the installation? Should the command `jupyter labextension install jupyter-matplotlib` install `rimraf` if it's not already present?

--- a/.gitignore
+++ b/.gitignore
@@ -5,9 +5,9 @@ build/
 *.py[cod]
 node_modules/
 
-# Compiled javascript
+# Included Javascript
 ipympl/static/
-ipympl/staticlab/
+ipympl/*.tgz
 
 # OS X
 .DS_Store

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,0 +1,34 @@
+language: python
+sudo: false
+dist: trusty
+addons:
+  firefox: latest
+cache:
+  directories:
+    - $HOME/.npm
+before_install:
+- wget https://github.com/mozilla/geckodriver/releases/download/v0.11.1/geckodriver-v0.11.1-linux64.tar.gz
+- mkdir geckodriver
+- tar -xzf geckodriver-v0.11.1-linux64.tar.gz -C geckodriver
+- export PATH=$PATH:$PWD/geckodriver
+- export DISPLAY=:99.0
+- sh -e /etc/init.d/xvfb start || true
+install:
+- wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
+- bash miniconda.sh -b -p $HOME/miniconda
+- export PATH="$HOME/miniconda/bin:$PATH"
+- hash -r
+- conda config --set always_yes yes --set changeps1 no
+- conda update -q conda
+- conda info -a
+- conda install nodejs notebook
+- pip install selenium cookiecutter
+- pip install --pre jupyterlab
+script:
+- cookiecutter . --no-input
+- cd jupyterlab_myextension
+- jupyter labextension install @jupyter-widgets/jupyterlab-manager --no-build
+- jupyter labextension install .
+- jupyter lab clean
+- jupyter labextension link .
+- python -m jupyterlab.selenium_check --NotebookApp.ip=127.0.0.1

--- a/.travis.yml
+++ b/.travis.yml
@@ -23,7 +23,7 @@ install:
 - conda info -a
 - conda install -c conda-forge nodejs notebook matplotlib ipywidgets
 - pip install --pre jupyterlab
-- pip install .
+- pip install -v .
 script:
 - jupyter nbextension install --py --symlink --sys-prefix ipympl
 - jupyter nbextension enable --py --sys-prefix ipympl

--- a/.travis.yml
+++ b/.travis.yml
@@ -21,14 +21,13 @@ install:
 - conda config --set always_yes yes --set changeps1 no
 - conda update -q conda
 - conda info -a
-- conda install nodejs notebook
-- pip install selenium cookiecutter
+- conda install -c conda-forge nodejs notebook matplotlib ipywidgets
 - pip install --pre jupyterlab
+- pip install .
 script:
-- cookiecutter . --no-input
-- cd jupyterlab_myextension
-- jupyter labextension install @jupyter-widgets/jupyterlab-manager --no-build
-- jupyter labextension install .
-- jupyter lab clean
-- jupyter labextension link .
+- jupyter nbextension install --py --symlink --sys-prefix ipympl
+- jupyter nbextension enable --py --sys-prefix ipympl
+- jupyter labextension list | grep -q jupyter-matplotlib
+- jupyter labextension install @jupyter-widgets/jupyterlab-manager
 - python -m jupyterlab.selenium_check --NotebookApp.ip=127.0.0.1
+- jupyter labextension link js

--- a/.travis.yml
+++ b/.travis.yml
@@ -22,12 +22,13 @@ install:
 - conda update -q conda
 - conda info -a
 - conda install -c conda-forge nodejs notebook matplotlib ipywidgets
+- pip install selenium
 - pip install --pre jupyterlab
 - pip install -v .
 script:
 - jupyter nbextension install --py --symlink --sys-prefix ipympl
 - jupyter nbextension enable --py --sys-prefix ipympl
-- jupyter labextension list | grep -q jupyter-matplotlib
+- jupyter labextension list 2>&1 | grep -q jupyter-matplotlib
 - jupyter labextension install @jupyter-widgets/jupyterlab-manager
 - python -m jupyterlab.selenium_check --NotebookApp.ip=127.0.0.1
 - jupyter labextension link js

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,2 +1,3 @@
 recursive-include ipympl/static *.*
+include ipympl/*.tgz
 include LICENSE

--- a/README.md
+++ b/README.md
@@ -21,24 +21,32 @@ Installation
 
 To install `ipympl` with conda:
 
-    $ conda install ipympl -c conda-forge
+    $ conda install -c conda-forge ipympl 
+    $ # If using the Notebook
+    $ conda install -c conda-forge widgetsnbextension
+    $ # If using JupyterLab
+    $ conda install nodejs
     $ jupyter labextension install @jupyter-widgets/jupyterlab-manager
 
 
 To install `ipympl` with pip:
 
     $ pip install ipympl
+    $ # If using the Notebook
     $ jupyter nbextension enable --py --sys-prefix ipympl
+    $ # If using JupyterLab
+    $ # Install nodejs: https://nodejs.org/en/download/
     $ jupyter labextension install @jupyter-widgets/jupyterlab-manager
 
 
-For a development installation (requires npm),
+For a development installation (requires node),
 
     $ git clone https://github.com/matplotlib/jupyter-matplotlib.git
     $ cd jupyter-matplotlib
     $ pip install -e .
     $ jupyter nbextension install --py --symlink --sys-prefix ipympl
     $ jupyter nbextension enable --py --sys-prefix ipympl
+    $ jupyter labextension install @jupyter-widgets/jupyterlab-manager --no-build
     $ jupyter labextension link ./js
     $ cd js && npm run watch  
     $ # Launch jupyterlab as `jupyter lab --watch` in another terminal

--- a/README.md
+++ b/README.md
@@ -22,14 +22,14 @@ Installation
 To install `ipympl` with conda:
 
     $ conda install ipympl -c conda-forge
-    $ jupyter labextension install jupyter-matplotlib
+    $ jupyter labextension install @jupyter-widgets/jupyterlab-manager
 
 
 To install `ipympl` with pip:
 
     $ pip install ipympl
     $ jupyter nbextension enable --py --sys-prefix ipympl
-    $ jupyter labextension install jupyter-matplotlib
+    $ jupyter labextension install @jupyter-widgets/jupyterlab-manager
 
 
 For a development installation (requires npm),

--- a/js/package.json
+++ b/js/package.json
@@ -12,7 +12,7 @@
   "scripts": {
     "build": "webpack",
     "clean": "rimraf dist/",
-    "postinstall": "npm run clean && npm run build",
+    "postinstall": "npm run build",
     "prepack": "npm run postinstall",
     "test": "echo \"Error: no test specified\" && exit 1",
     "watch": "webpack --watch"
@@ -28,8 +28,7 @@
   },
   "dependencies": {
     "jquery": "^2.1.4",
-    "@jupyter-widgets/base": "^1.0.0",
-    "@jupyter-widgets/jupyterlab-manager": "^0.28.0",
+    "@jupyter-widgets/base": "^1.1.0",
     "lodash": "^4.17.4"
   }
 }

--- a/js/package.json
+++ b/js/package.json
@@ -27,8 +27,9 @@
     "webpack": "^3.5.5"
   },
   "dependencies": {
-    "jquery": "^2.1.4",
     "@jupyter-widgets/base": "^1.1.0",
+    "jquery": "^2.1.4",
+    "jquery-ui": "^1.12.1",
     "lodash": "^4.17.4"
   }
 }

--- a/js/src/labplugin.js
+++ b/js/src/labplugin.js
@@ -1,10 +1,10 @@
 var jupyter_matplotlib = require('./index');
 
-var jupyterlab_widgets = require('@jupyter-widgets/jupyterlab-manager');
+var jupyter_widgets = require('@jupyter-widgets/base');
 
 module.exports = {
     id: 'matplotlib-jupyter:main',
-    requires: [jupyterlab_widgets.INBWidgetExtension],
+    requires: [jupyterlab_widgets.IJupyterWidgetRegistry],
     activate: function(app, widgets) {
         widgets.registerWidget({
             name: 'jupyter-matplotlib',

--- a/setup.py
+++ b/setup.py
@@ -91,7 +91,7 @@ class NPM(Command):
         return self.has_npm() and not node_modules_exists
 
     def should_run_npm_pack(self):
-        files = glob.glob()
+        files = glob.glob(tar_path)
         tarball_exists = len(files) > 0
         return self.has_npm() and not tarball_exists
 
@@ -110,7 +110,7 @@ class NPM(Command):
 
         if self.should_run_npm_pack():
             check_call(['npm', 'pack'], cwd=node_root, stdout=sys.stdout, stderr=sys.stderr)
-            target = glob.glob(pjoin(here, 'js', '*.tgz'))[0]
+            target = glob.glob(pjoin(tar_path))[0]
             shutil.move(target, pjoin(here, 'ipympl'))
 
         files = glob.glob(tar_path)
@@ -125,6 +125,7 @@ class NPM(Command):
 
         # update package data in case this created new files
         update_package_data(self.distribution)
+
 
 version_ns = {}
 with open(pjoin(here, 'ipympl', '_version.py')) as f:

--- a/setup.py
+++ b/setup.py
@@ -109,9 +109,7 @@ class NPM(Command):
             os.utime(self.node_modules, None)
 
         if self.should_run_npm_pack():
-            check_call(['npm', 'pack'], cwd=node_root, stdout=sys.stdout, stderr=sys.stderr)
-            target = glob.glob(pjoin(tar_path))[0]
-            shutil.move(target, pjoin(here, 'ipympl'))
+            check_call(['npm', 'pack', node_root], cwd=pjoin(here, 'ipympl'), stdout=sys.stdout, stderr=sys.stderr)
 
         files = glob.glob(tar_path)
         self.targets.append(tar_path if not files else files[0])

--- a/setup.py
+++ b/setup.py
@@ -4,16 +4,19 @@ from setuptools.command.sdist import sdist
 from setuptools.command.build_py import build_py
 from setuptools.command.egg_info import egg_info
 from subprocess import check_call
+import glob
 import os
+import shutil
 import sys
-import platform
+from os.path import join as pjoin
 
 here = os.path.dirname(os.path.abspath(__file__))
-node_root = os.path.join(here, 'js')
-is_repo = os.path.exists(os.path.join(here, '.git'))
+node_root = pjoin(here, 'js')
+is_repo = os.path.exists(pjoin(here, '.git'))
+tar_path = pjoin(here, 'ipympl', '*.tgz')
 
 npm_path = os.pathsep.join([
-    os.path.join(node_root, 'node_modules', '.bin'),
+    pjoin(node_root, 'node_modules', '.bin'),
                 os.environ.get('PATH', os.defpath),
 ])
 
@@ -62,12 +65,12 @@ class NPM(Command):
 
     user_options = []
 
-    node_modules = os.path.join(node_root, 'node_modules')
+    node_modules = pjoin(node_root, 'node_modules')
 
     targets = [
-        os.path.join(here, 'ipympl', 'static', 'extension.js'),
-        os.path.join(here, 'ipympl', 'static', 'index.js'),
-        os.path.join(here, 'ipympl', 'static', 'package.json')
+        pjoin(here, 'ipympl', 'static', 'extension.js'),
+        pjoin(here, 'ipympl', 'static', 'index.js'),
+        pjoin(here, 'ipympl', 'static', 'package.json')
     ]
 
     def initialize_options(self):
@@ -80,13 +83,17 @@ class NPM(Command):
         try:
             check_call(['npm', '--version'])
             return True
-        except:
+        except Exception:
             return False
 
     def should_run_npm_install(self):
-        package_json = os.path.join(node_root, 'package.json')
         node_modules_exists = os.path.exists(self.node_modules)
-        return self.has_npm()
+        return self.has_npm() and not node_modules_exists
+
+    def should_run_npm_pack(self):
+        files = glob.glob()
+        tarball_exists = len(files) > 0
+        return self.has_npm() and not tarball_exists
 
     def run(self):
         has_npm = self.has_npm()
@@ -101,6 +108,14 @@ class NPM(Command):
             check_call(['npm', 'install'], cwd=node_root, stdout=sys.stdout, stderr=sys.stderr)
             os.utime(self.node_modules, None)
 
+        if self.should_run_npm_pack():
+            check_call(['npm', 'pack'], cwd=node_root, stdout=sys.stdout, stderr=sys.stderr)
+            target = glob.glob(pjoin(here, 'js', '*.tgz'))[0]
+            shutil.move(target, pjoin(here, 'ipympl'))
+
+        files = glob.glob(tar_path)
+        self.targets.append(tar_path if not files else files[0])
+
         for t in self.targets:
             if not os.path.exists(t):
                 msg = 'Missing file: %s' % t
@@ -112,7 +127,7 @@ class NPM(Command):
         update_package_data(self.distribution)
 
 version_ns = {}
-with open(os.path.join(here, 'ipympl', '_version.py')) as f:
+with open(pjoin(here, 'ipympl', '_version.py')) as f:
     exec(f.read(), {}, version_ns)
 
 setup_args = {
@@ -128,6 +143,9 @@ setup_args = {
             'ipympl/static/index.js',
             'ipympl/static/index.js.map',
             'ipympl/static/package.json'
+        ]),
+        ('share/jupyter/lab/extensions', [
+            os.path.relpath(f, '.') for f in glob.glob(tar_path)
         ])
     ],
     'install_requires': [

